core/dive.c

```diff
@@ -528,14 +528,8 @@ static void copy_tl(struct tag_entry *st, struct tag_entry *dt)
 		*_dptr = 0;                              \
 	}
 
-/* copy_dive makes duplicates of many components of a dive;
- * in order not to leak memory, we need to free those .
- * copy_dive doesn't play with the divetrip and forward/backward pointers
- * so we can ignore those */
-void clear_dive(struct dive *d)
+static void free_dive_structures(struct dive *d)
 {
-	if (!d)
-		return;
```

Comment:
why not leave this in there? If you call free_dive_structures(NULL) bad things happen. We may not be at risk of doing this today, but still... it seems cheap insurance...

Reply:
Yes, that makes sense...

With this perhaps really wait until release? Seems correct, but nevertheless a bit scary.

Reply:
hehe... you're not paying attention. The release is out :-)

Reply:
Oops. I admit I didn't follow too closely as there was too much activity.

Reply:
No problem - I'm just giving you a hard time.
I really want BLE on Windows to work... and I think that's where most of the activity comes from.